zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt

```diff
@@ -36,25 +38,48 @@ data class LoadedManifest(
     val freshManifest = manifest.copy(
       freshAtEpochMs = freshAtEpochMs,
     )
-    val freshManifestBytes = jsonForManifest.encodeToString(freshManifest).encodeUtf8()
+    val freshManifestBytes = freshManifest.encodeToString().encodeUtf8()
     return LoadedManifest(freshManifestBytes, freshManifest, freshAtEpochMs)
   }
 }
 
-internal val jsonForManifest = Json {
+private val jsonForManifest = Json {
   // For backwards-compatibility, allow new fields to be introduced.
   ignoreUnknownKeys = true
 
   // Because new releases may change default values, it's best to encode them.
   encodeDefaults = true
 }
 
+/**
+ * Confirm the manifest is of a reasonable size before proceeding to operate on it. 10 KiB is a
+ * typical size for our test applications. 640K ought to be enough for anybody.
+ */
+const val MANIFEST_MAX_SIZE = 640 * 1024
```

Comment:
Meant to be public?

Reply:
Yeah, we’re using it from the CLI module. I still don’t really have good ideas for internal APIs that can’t be internal visibility

Reply:
There are friend paths which make `internal` work across compilations, but it's complicated to set up. It's how test sources see main source's `internal` things today.

This is the feature request for the good UX version: https://youtrack.jetbrains.com/issue/KT-22284/